modules/packages/apk

```diff
@@ -0,0 +1,110 @@
+#!/bin/sh -e
+get_package_data() {
+  file="${INPUT_File?File must be given to get-package-data}"
+  set +e
+  apk verify $file 2>/dev/null >/dev/null
+  rc=$?
+  set -e
+  if [ 0 = $rc ]; then
+    echo "PackageType=file"
+    name=$(basename $file)
+    echo $name | sed -e 's/-/ /' -e 's/.apk/ apk/' | awk '
+{
+  printf("Name=%s\n",$1)
+  printf("Version=%s\n",$2)
+}'
+  else
+    echo PackageType=repo
+    echo Name=$file
+  fi
+}
+
+list_installed() {
+  # Example `apk list --installed` output:
+  # busybox-1.32.0-r3 x86_64 {busybox} (GPL-2.0-only) [installed]
+  #
+  # After rewrite:
+  # Name=busybox
+  # Version=1.32.0-r3
+  # Architecture=x86_64
+  apk list --installed | sed 's/-/ /' | awk '
+{
+    printf("Name=%s\n",$1)
+    printf("Version=%s\n",$2)
+    printf("Architecture=%s\n",$3)
+}'
+}
+
+repo_install() {
+  name="${INPUT_Name?Name must be given to repo-install}"
+  version="${INPUT_Version}"
+  if [ ! -z "${INPUT_Version}" ]; then
+    apk add "$name=$version" 2>/dev/null >/dev/null
+  else
+    apk add --upgrade "$name" 2>/dev/null >/dev/null
+  fi
+}
+
+file_install() {
+  file="${INPUT_File?File must be given to file-install}"
+  apk add $file 2>/dev/null >/dev/null
```

Comment:
cool beans

Reply:
CFE-3451, in title and will be in squashed single commit in a moment.